### DISCUSSION TITLE
x-data inside x-for problem

### DISCUSSION THREAD

**Discussion by mccarlosen** [alpinejs/alpine#871] [4. General]
I have a problem. I don't know how to approach it with AlpineJS.

- I want the loading icons to be independent
- I want to access x-for items from nested x-data
- I want each item to be a component

[Example sandbox](https://codesandbox.io/s/nervous-lamarr-9q4lx?file=/src/styles.css)

```html
<template x-for="(item, index) in data" :key="index">
  <div x-data="{ loading: false }">
    <!-- item is not accessible -->
    <a @click="download(item.id)">
      <!-- item is not accessible -->
      <i x-show="loading" class="animate-spin fas fa-spinner></i> <span x-text="item.title"></span>
    </a>
  </div>
</template>
```

Any idea, another way to do it.

**Comment by KevinBatdorf** [comment]
There's not a built in way, but there are some alternative ideas on[ this issue](https://github.com/alpinejs/alpine/issues/793) you can look at.

**Reply by mccarlosen** [reply]
Excellent.!! Thanks
